### DISCUSSION TITLE
using Jinja template AND set_cookie() for post request

### DISCUSSION THREAD

**Discussion by fcabaud** [fastapi/fastapi#6883] [Questions] [answered]
### First check

* [*] I added a very descriptive title to this issue.
* [*] I used the GitHub search to find a similar issue and didn't find it.
* [*] I searched the FastAPI documentation, with the integrated search.
* [*] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [*] I already read and followed all the tutorial in the docs and didn't find an answer.
* [*] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [*] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [*] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [*] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI

app = FastAPI()
templates = Jinja2Templates(directory='templates/')

@app.post('/select_project_post')
def select_project_post(response: Response, request: Request, project: str = Form(...)):
        response.set_cookie(key='cookie_current_project_name_no_extension', value=project, httponly=True)
        return templates.TemplateResponse('base.html', context={'response': response, 'request': request, 'project': project} )
```

### Description

* Open the browser, select a file a do a post request.
* The template base.html page is correctly displayed but the cookie is not set. It's not set due to 
```return templates.TemplateResponse('base.html', context={'response': response, 'request': request, 'project': project} )
```
if I return
```return {"selected project": project}```
then the cookie is set.

My question is **How setting cookie AND using a Jinja2 template for a post request ?**
I didnt find the answer to this question in [https://fastapi.tiangolo.com/advanced/response-cookies/](url)

### Environment

* OS: linux:
* FastAPI Version 0.63.0:
* Python version: 3.8.6
* Jinja2 version: 2.11.3

**Comment by krishnardt** [comment, marked as answer]
You can use the following code to set the cookie

```
response = templates.TemplateResponse('sample_html.html', context={'request': request, 'project': project} )
response.set_cookie(key='cookie_current_project_name_no_extension', value=project, httponly=True)
return response
```

you will get the cookie in response

`{'body_iterator': <async_generator object BaseHTTPMiddleware.call_next.<locals>.body_stream at 0x00000260521F5288>, 'status_code': 200, 'media_type': None, 'background': None, 'raw_headers': [(b'content-length', b'686'), (b'content-type', b'text/html; charset=utf-8'), **(b'set-cookie', b'cookie_current_project_name_no_extension=str**; HttpOnly; Path=/; SameSite=lax')]}
`

**Comment by fcabaud** [comment]
Thank you so much  krishnardt !!
You made my day.

**Comment by tiangolo** [comment]
Thanks for the help here @krishnardt ! 👏  🙇 

Thanks for reporting back and closing the issue @fcabaud  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
